native_client/BUILD

```diff
@@ -361,24 +393,13 @@ cc_binary(
     ],
     deps = [":decoder", ":coqui_stt_bundle"],
     copts = ["-std=c++14", "-fno-exceptions", "-fwrapv", "-pthread"],
-    linkopts = DECODER_LINKOPTS + [
-        "--bind",
-        "-sWASM=1",
-        # Allow to grow memory allocation when needed (for example for loading models).
-        "-sALLOW_MEMORY_GROWTH",
-        "-sMAXIMUM_MEMORY=4GB",
-        "-sMALLOC=emmalloc",
-        # We need to specify a threadpool size, otherwise TFLite will just do nothing
-        # and deadlock.
-        "-sPTHREAD_POOL_SIZE=4",
-        # This is a library, so no 'main' is expected.
-        "--no-entry",
-        # Experiencing problems? Uncomment the flags below to have better
-        # error messages and sanity checks.
-        # "-sASSERTIONS=2",
-        # "-g3",
-        # "-gsource-map",
-    ],
+    linkopts = select({
+        ":wasm_es6": DECODER_LINKOPTS_WASM + [
+            "-sEXPORT_ES6=1",
+            "-sSINGLE_FILE=1",
```

Comment:
Would you kindly add a comment here explaining why, even though we have `SINGLE_FILE` here, it's producing multiple files? (because of the "pthread" story you mentioned, right?)

Reply:
`SINGLE_FILE` exists since before pthread support and it currently only inlines the wasm bytecode as a base64 string inside the main JS. Pthreads are implemented as separate (smaller) workers which actually load the main JS from within. There are some requests to inline or merge both files when using SINGLE_FILE (see [emscripten#9796](https://github.com/emscripten-core/emscripten/issues/9796) and [emscripten#14329](https://github.com/emscripten-core/emscripten/issues/14329#issuecomment-852522630))

Generally, as noted in the references above, using `SINGLE_FILE` generates a large asset (larger than the sum of both `stt_wasm.js`, which means longer download and startup times. Blinded by the original goal of making it available via CDN, I thought it would be necessary to inline the wasm binary for ease of use as a library, but I just tested it and webpack can handle the separate asset perfectly (rollup should also be fine; I am not sure about parcel, but I guess it'd also work.)

I will thus remove this flag. At this point we could consider dropping es5 support (ES6 is necessary for bundlers, see [emscripten#14135](https://github.com/emscripten-core/emscripten/pull/14135)) but ES6 imports are still not supported in workers under Firefox ([reference](https://bugzilla.mozilla.org/show_bug.cgi?id=1540913)) so I would still add it to the github releases.

Reply:
Awesome, I agree with what you outlined. Thank you for the thorough explaination !